utils/Azure/acrTools.ts

```diff
@@ -111,24 +113,29 @@ export async function acquireACRAccessTokenFromRegistry(registry: Registry, scop
 /** Obtains refresh and access tokens for Azure Active Directory. */
 export async function acquireAADTokens(session: AzureSession): Promise<{ aadAccessToken: string, aadRefreshToken: string }> {
     return new Promise<{ aadAccessToken: string, aadRefreshToken: string }>((resolve, reject) => {
-        const credentials: any = session.credentials;
-        const environment: any = session.environment;
-        credentials.context.acquireToken(environment.activeDirectoryResourceId, credentials.username, credentials.clientId, (err: any, result: any) => {
-            if (err) {
-                reject(err);
-            } else {
-                resolve({
-                    aadAccessToken: result.accessToken,
-                    aadRefreshToken: result.refreshToken,
-                });
-            }
-        });
+        const credentials = <{ context: AuthenticationContext, username: string, clientId: string }><any>session.credentials;
```

Comment:
Is there a way around using the `<any>` casting here? 

From a read, my understanding, it looks like session.credentials is not guaranteed to be of the specified type, so casting it to <any> helps the other cast.